### PR TITLE
Better collisions!

### PR DESCRIPTION
**Pull request by ChielBruin**
Implemented pixel-perfect collision.
Also added tests!

### CODE REVIEW

**Comment by Diocruel** [comment]
Big pull request. I tested the branch earlier. Works for me

File: src/main/java/com/github/fishio/PlayingField.java [hunk 1]
```diff
@@ -197,11 +197,12 @@ public void redraw() {
 
 	/**
 	 * Checks for collisions.
+	 * Only needed for the players on movables
 	 */
 	public void checkCollisions() {
```
Comment (ChielBruin) [inline src/main/java/com/github/fishio/PlayingField.java]: rename to checkPlayerCollisions

File: src/main/java/com/github/fishio/Preloader.java [hunk 2]
```diff
@@ -137,6 +159,59 @@ public static Image getImageOrLoad(String file) {
...
+	 * @param file
+	 * 		the file of the Image.
+	 * 
+	 * @return
+	 * 		the alpha data of the image
+	 */
+	public static boolean[][] getAlphaDataOrLoad(String file) {
+		boolean[][] data;
+		synchronized (IMAGE_DATA) {
+		data = IMAGE_DATA.get(file);
```
Comment (ChielBruin) [inline src/main/java/com/github/fishio/Preloader.java]: missing tab
<layout

> layout

File: src/test/java/com/github/fishio/TestCollisionMask.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+		boolean[][] data = CollisionMask.buildData(image);
+		ca = new CollisionMask(new Vec2d(0, 0), 4, 4, data, 1);
+	}
+
+	/**
+	 * Test for {@link CollisionMask#buildData(Image).
+	 */
+	@Test
+	public void testBuildData() {
+		boolean[][] exp = {{true, false, false},
```
Comment (ChielBruin) [inline src/test/java/com/github/fishio/TestCollisionMask.java:39]: Indentation

<layout

> layout

**Comment by Taeir** [comment]
After fixing the comments, the pull request can be made.

**Comment by Diocruel** [comment]
Should be good to merge now then.

**Comment by Taeir** [comment]
Good to merge :)

**Pull request merged**